run-convert2h5/convert_edepsim_roottoh5.py

```diff
@@ -405,28 +405,29 @@ def dump(input_file, output_file, keep_all_dets=False):
                 segment[iHit]["segment_id"] = segment_id
                 segment_id += 1
                 try:
-                    segment[iHit]["traj_id"] = trackMap[hitSegment.Contrib[0]]
-                    seg_local_traj_id = hitSegment.Contrib[0]
+                    segment[iHit]["traj_id"] = hitSegment.Contrib[0]
+                    seg_traj_id = hitSegment.Contrib[0]
                     if segment[iHit]["traj_id"] not in trajectories["traj_id"]:
                         # Given event.Trajectories is ordered by traj_id (trajectory.GetTrackId())
-                        trajectory = event.Trajectories[seg_local_traj_id]
+                        trajectory = event.Trajectories[seg_traj_id]
+                        # Trace back in the family tree
                         while trajectory.GetParentId() >= -1:
-                            if trackMap[trajectory.GetTrackId()] in trajectories["traj_id"]:
+                            if trajectory.GetTrackId() in trajectories["traj_id"]:
```

Comment:
Should this be:

```python
if trackMap[trajectory.GetTrackId()] in trajectories["file_traj_id"]:
```

since `traj_id` is no longer unique in `trajectories`?

Or if we want to avoid using `file_traj_id`, then shouldn't we check for an element of `trajectories` that matches both the `vertex_id` and the `traj_id`?

Reply:
nevermind, `traj_id` is unique in `trajectories` because `trajectories` is reinitialized for every edepsim event (i.e. vertex)